### PR TITLE
upgrade secp256kfun

### PR DESCRIPTION
**Pull request by delta1**
re: #1459  

upgrades the secp256kfun dependency, the previously pinned version fails to compile on rust 1.74 

fixes a few resulting issues, and some minor cleanup refactoring 

also adds a new CI job to run cargo check on rust stable 

also closes: 
- #1410 
- #1412

### CODE REVIEW

**Comment by binarybaron** [comment]
I will need some time to review this as this is quite critical code and I'm not too familiar with the library.

**Comment by ikmckenz** [comment]
Out of curiosity, why resolve on 0.24.1 instead of the most recent release (0.28)?

**Comment by delta1** [comment]
> Out of curiosity, why resolve on 0.24.1 instead of the most recent release (0.28)?

looks like you’re referring to secp256k1, their PR is upgrading secp256kfun. as for the versions, from the lockfile it looks like bitset still has a dependency on 0.24 of secp256k1, and secp256kfun is still depending on 0.27 of secp256k1

File: swap/src/bitcoin.rs [hunk 1]
```diff
@@ -206,20 +206,21 @@ pub fn verify_encsig(
...
-    let miniscript = MINISCRIPT_TEMPLATE.replace('A', &A).replace('B', &B);
+    let miniscript = MINISCRIPT_TEMPLATE
+        .replace('A', &A.to_string())
+        .replace('B', &B.to_string());
 
     let miniscript =
         bdk::miniscript::Miniscript::<bitcoin::PublicKey, Segwitv0>::from_str(&miniscript)
             .expect("a valid miniscript");
 
-    Descriptor::Wsh(Wsh::new(miniscript).expect("a valid descriptor"))
```
Comment (binarybaron) [inline swap/src/bitcoin.rs:222]: why are we removing this expect(...) here?

**Comment by binarybaron** [comment]
LGTM but have you completed a swap using the new CLI version and an old ASB version yet?

**Comment by delta1** [comment]
updated to include #1546 #1547

**Comment by delta1** [comment]
updated to include #1549 #1550

**Comment by delta1** [comment]
update to include #1556 and #1557 

also sets the revision in the Cargo.toml manifest

**Comment by binarybaron** [comment]
Is this 100% backwards compatible in terms of the database and p2p protocols?

**Comment by delta1** [comment]
feel free to test it

**Comment by delta1** [comment]
@ikmckenz if you have a moment, please try test build this branch asb/swap, and gets those binaries on testnet performing a swap against the previous version

**Comment by delta1** [comment]
Sigh I don't have any testnet bitcoin to test with :weary: @binarybaron any donations? 

`tb1qttry3wfcwcf7gyv26a5elt9hcran7nsnfj9zgj`

**Comment by delta1** [comment]
> Is this 100% backwards compatible in terms of the database and p2p protocols?

tested a number of testnet swaps between this version of asb and latest swap release, as well as this version of swap with latest asb release 

happy for this change to be a version change to 0.13

**Comment by linsui** [comment]
Why do you use the git repo instead of 0.10.0 on crates.io?

**Comment by delta1** [comment]
> Why do you use the git repo instead of 0.10.0 on crates.io?

good question, i think that’s a hangover from where we were ahead of their releases to crates.io 

i’m going to rework this PR to use the crates version, thanks for the suggestion

**Review by binarybaron — APPROVED** [review]
LGTM!

**Comment by delta1** [comment]
includes a fix for #1645

**Pull request merged**